sources/src/test/java/com/google/solutions/jitaccess/core/catalog/project/TestMpaProjectRoleCatalog.java

```diff
@@ -35,615 +36,727 @@
-        eq(EnumSet.of(Entitlement.Status.AVAILABLE))))
-      .thenReturn(EntitlementSet.empty());
+        .findRequesterPrivileges(
+            eq(SAMPLE_REQUESTING_USER),
+            eq(SAMPLE_PROJECT),
+            eq(Set.of(peerApproval)),
+            eq(EnumSet.of(RequesterPrivilege.Status.AVAILABLE))))
+        .thenReturn(RequesterPrivilegeSet.empty());
 
     assertThrows(
-      AccessDeniedException.class,
-      () -> catalog.listReviewers(
-        SAMPLE_REQUESTING_USER,
-        new ProjectRoleBinding(new RoleBinding(SAMPLE_PROJECT, SAMPLE_ROLE))));
+        AccessDeniedException.class,
+        () -> catalog.verifyUserCanActivateRequesterPrivileges(
+            SAMPLE_REQUESTING_USER,
+            SAMPLE_PROJECT,
+            peerApproval,
+            List.of(peerApprovalPrivilege.id())));
   }
 
   @Test
-  public void whenUserAllowedToActivateEntitlement_ThenListReviewersExcludesUser() throws Exception {
+  public void whenUserHasTopicEmpty_ThenVerifyUserCanActivateRequesterPrivilegesReturns()
+      throws Exception {
     var policyAnalyzer = Mockito.mock(PolicyAnalyzerRepository.class);
 
     var catalog = new MpaProjectRoleCatalog(
-      policyAnalyzer,
-      Mockito.mock(ResourceManagerClient.class),
-      new MpaProjectRoleCatalog.Options(null, Duration.ofMinutes(30), 1, 2));
+        policyAnalyzer,
+        Mockito.mock(ResourceManagerClient.class),
+        new MpaProjectRoleCatalog.Options(null, Duration.ofMinutes(30), 1, 2));
+
+    var peerApprovalPrivilege = new RequesterPrivilege<>(
+        new ProjectRoleBinding(new RoleBinding(SAMPLE_PROJECT, SAMPLE_ROLE)),
+        "-",
+        new PeerApproval(""),
+        RequesterPrivilege.Status.AVAILABLE);
 
-    var mpaEntitlement = new Entitlement<>(
-      new ProjectRoleBinding(new RoleBinding(SAMPLE_PROJECT, SAMPLE_ROLE)),
-      "-",
-      ActivationType.MPA,
-      Entitlement.Status.AVAILABLE);
+    var peerApproval = new PeerApproval("topic");
 
     when(policyAnalyzer
-      .findEntitlements(
-        eq(SAMPLE_REQUESTING_USER),
-        eq(SAMPLE_PROJECT),
-        eq(EnumSet.of(ActivationType.MPA)),
-        eq(EnumSet.of(Entitlement.Status.AVAILABLE))))
-      .thenReturn(new EntitlementSet<>(
-        new TreeSet<>(Set.of(mpaEntitlement)),
-        Set.of(),
-        Set.of()));
+        .findRequesterPrivileges(
+            eq(SAMPLE_REQUESTING_USER),
+            eq(SAMPLE_PROJECT),
+            eq(Set.of(peerApproval)),
+            eq(EnumSet.of(RequesterPrivilege.Status.AVAILABLE))))
+        .thenReturn(new RequesterPrivilegeSet(Set.of(peerApprovalPrivilege), Set.of(), Set.of()));
+
+    assertDoesNotThrow(
+        () -> catalog.verifyUserCanActivateRequesterPrivileges(
+            SAMPLE_REQUESTING_USER,
+            SAMPLE_PROJECT,
+            peerApproval,
+            List.of(peerApprovalPrivilege.id())));
+  }
+
+  @Test
+  public void whenRequestHasTopicEmpty_ThenVerifyUserCanActivateRequesterPrivilegesThrows()
```

Comment:
Something is confusing with the naming here, because the previous test is called the same thing but ends in not throwing, but this test ends in throwing, but the "when" sounds to be exactly the same. let's discuss

Reply:
One is named `whenUserHasTopicEmpty` whereas the other is `whenRequestHasTopicEmpty`. This test tests that a user with `RequesterPrivilege` of type `PeerApproval("topic")` **cannot** activate a RequesterPrivilege of type `PeerApproval("")`.

Reply:
Understood